src/bun.js/bindings/ImportMetaObject.cpp

```diff
@@ -403,6 +403,11 @@ JSC_DEFINE_CUSTOM_GETTER(jsImportMetaObjectGetter_require, (JSGlobalObject * glo
 
     return JSValue::encode(thisObject->requireProperty.getInitializedOnMainThread(thisObject));
 }
+JSC_DEFINE_CUSTOM_GETTER(jsImportMetaObjectGetter_env, (JSGlobalObject * jsGlobalObject, JSC::EncodedJSValue thisValue, PropertyName propertyName))
+{
```

Comment:
Can you add a test that runs this in node:vm? Just to make sure this jsCast doesn’t crash when the global object is not a Zig::GlobalObject

Reply:
Tried to run this test, but the parser complains with `SyntaxError: import.meta is only valid inside modules.`, and I couldn't find any way to force it to run as an ESM, other than using node:vm modules which are not yet implemented.

Reply:
you could pass the import.meta.env object to the node:vm instance